README.md

```diff
@@ -80,7 +80,58 @@ Alternatively, you can specify `html.root` in your configuration and this will b
 
 You can add or override the default [webpack-isomorphic-tools](https://github.com/halt-hammerzeit/webpack-isomorphic-tools) configuration, by providing a `toolsConfigPath` value to your `config.js`.
 
-#### Scripts
+### Plugins
+
+Hooks can be used in order to override the default implementation of certain functions. Files that define hooks are called plugins and must be listed in the `plugins` property of `config/universal-redux.config.js`. The plugin paths are resolved as follow:
+
+ - ${project-src}/{path}
+ - ${project-root}/node_modules/universal-redux-plugin-{path}
+ - ${project-root}/node_modules/{path}
+
+The plugin file should register the implementations of the functions it wants to override as follow:
+
+```js
+import { hooks, register } from 'universal-redux/lib/hooks';
+
+register(hooks.client.GENERATE_ROOT_COMPONENT, () => {
+  //Generate the component
+  return { mycomponent };
+});
+```
+
+> If multiple plugins register the same hook, the last plugin to register is takes precedence.
+
+#### client.GENERATE_ROOT_COMPONENT
+
+This hook can be used to override the default rendering of the root component on the client.
+
+##### Arguments
+
+The hook implementation is provided a single `params` object with the following properties
+ - **store** - The redux store
+ - **routes** - The routes exported by your `routes.js` file
+ - **history** - The `react-router` history
+ - **devComponent** *(optional)* - A react component that renders the dev tools
+
+##### Return
+
+The hook implementation should return a valid react component or a promise that resolves to a valid react component.
```

Comment:
I quickly googled it and I believe the answer is React element. I should be an instantiated React Component. I'll clarify it.